src/controller/artifact/annotation/controller.go

```diff
@@ -0,0 +1,44 @@
+package annotation
+
+import (
+	"fmt"
+
+	"github.com/goharbor/harbor/src/controller/artifact/annotation/checker"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type AnnotationController interface {
+	ProcessAnnotations(artifact *artifact.Artifact, manifest *v1.Manifest, metadata map[string]interface{}) error
```

Comment:
If it does not have multiple implementations, maybe we can define it as a struct?

Reply:
It doesn't matter. Use interface to separate different module is a good convention.